src/core/libmaven/jsonReports.cpp

```diff
@@ -1,286 +1,674 @@
+            REQUIRE( mMzInput == doctest::Approx( mMzSaved ).epsilon(0.05));
+            REQUIRE( bMzInput == doctest::Approx( bMzSaved ).epsilon(0.05));
+            REQUIRE( mzminInput == doctest::Approx( mzminSaved ).epsilon(0.05));
+            REQUIRE( mzmaxInput == doctest::Approx( mzmaxSaved ).epsilon(0.05));
+            REQUIRE( rtInput == doctest::Approx( rtSaved ).epsilon(0.05));
+            REQUIRE( rtminInput == doctest::Approx( rtminSaved ).epsilon(0.05));
+            REQUIRE( rtmaxInput == doctest::Approx( rtmaxSaved ).epsilon(0.05));
+            REQUIRE( qualityInput == doctest::Approx( qualitySaved ).epsilon(0.05));
+            REQUIRE( pIInput == doctest::Approx( pISaved ).epsilon(0.05));
+            REQUIRE( pBLLInput == doctest::Approx( pBLLSaved ).epsilon(0.05));
+            REQUIRE( pAreaInput == doctest::Approx( pAreaSaved ).epsilon(0.05));
+            REQUIRE( pSAreaInput == doctest::Approx( pSAreaSaved ).epsilon(0.05));
+            REQUIRE( pATopInput == doctest::Approx( pATopSaved ).epsilon(0.05));
+            REQUIRE( pANotCorInput == doctest::Approx( pANotCorSaved ).epsilon(0.05));
+            REQUIRE( pATNOtSaved == doctest::Approx( pATNotInput ).epsilon(0.05));
+            REQUIRE( NoNoiseInput == doctest::Approx( NoNoiseSaved ).epsilon(0.05));
+            REQUIRE( SBRatioInput == doctest::Approx( SBRatioSaved ).epsilon(0.05));
+            REQUIRE( fBankSInput == doctest::Approx( fBankSSaved ).epsilon(0.05));
+            REQUIRE( pAFInput == doctest::Approx( pAFSaved ).epsilon(0.05));
+            REQUIRE( symmetryInput == doctest::Approx( symmetrySaved ).epsilon(0.05));
+            REQUIRE( noNFracInput == doctest::Approx( noNFracSaved ).epsilon(0.05));
+            REQUIRE( gOFracInput == doctest::Approx( gOFracSaved ).epsilon(0.05));
+            REQUIRE( gOverlapInput == doctest::Approx( gOverlapSaved ).epsilon(0.05));
+            REQUIRE( gFitInput == doctest::Approx( gFitSaved ).epsilon(0.05));
+            REQUIRE( pRankInput == doctest::Approx( pRankSaved ).epsilon(0.05));
+            REQUIRE( pWidthInput == doctest::Approx( pWidthSaved ).epsilon(0.05));
+            
+            for(size_t l = 0 ; l < rootInput["groups"][input]["peak"][i]["eic"]["rt"].size() ; l++)
+                REQUIRE(rootInput["groups"][input]["peak"][i]["eic"]["rt"][l] ==
+                                                                rootSaved["groups"][saved]["peak"][i]["eic"]["rt"][l]);
+                                                                                              
+            for(size_t l = 0; i < rootInput["groups"][input]["peak"][i]["eic"]["intensity"].size() ; l++)
+                REQUIRE(rootInput["groups"][input]["peak"][i]["eic"]["intensity"][l] == 
+                                                                rootSaved["groups"][saved]["peak"][i]["eic"]["intensity"][l]);
+                                
+        }
+    }

```

Comment:
I think, at this point, you should delete the "test.json" file that was created in the beginning of this function.